src/ts/index.ts

```diff
@@ -238,17 +262,21 @@ export class SettlementEncoder {
    * Additionally, if the order references new tokens that the encoder has not
    * yet seen, they are added to the tokens array.
    * @param order The order of the trade to encode.
-   * @param executedAmount The executed trade amount.
+   * @param trade The execution details for the trade.
    * @param signature The signature for the order data.
    * @param scheme The signing scheme to used to generate the specified
    * signature. See {@link SigningScheme} for more details.
    */
   public encodeTrade(
     order: Order,
-    executedAmount: BigNumberish,
+    trade: Partial<TradeExecution>,
```

Comment:
I'd prefer this to be called `executionParameters`, since a trade is something else in this code. (The same naming convention is used in another function below.)